extensions/stackdriver/stackdriver.cc

```diff
@@ -472,6 +497,48 @@ StackdriverRootContext* StackdriverContext::getRootContext() {
   return dynamic_cast<StackdriverRootContext*>(root);
 }
 
+void StackdriverRootContext::logTCPOpenOnTimeout(
+    StackdriverRootContext::TcpRecordInfo& record_info,
+    const std::string& destination_namespace, bool outbound,
+    const ::Wasm::Common::FlatNode& peer_node) {
+  auto cur = static_cast<long int>(
+      proxy_wasm::null_plugin::getCurrentTimeNanoseconds());
+  ::Wasm::Common::RequestInfo& request_info = *record_info.request_info;
+  // Attempt to log TCP Open Event if tcp open request log has timed out and it
+  // has not been logged before.
+  if (enableServerAccessLog() && !record_info.tcp_open_log_entry_logged &&
+      ((cur - request_info.start_time) > tcp_log_entry_timeout_)) {
+    // Populate RequestInfo with local node information.
+    ::Wasm::Common::populateTCPRequestInfo(
+        outbound, record_info.request_info.get(), destination_namespace);
+    ::Wasm::Common::populateExtendedRequestInfo(record_info.request_info.get());
+    logTCPOpen(record_info, peer_node);
+    // Set is_populated to false, so that populateTCPRequestInfo can be called
+    // again when peer metadata is available by RecordTCP.
+    record_info.request_info->is_populated = false;
+  }
+}
+
+void StackdriverRootContext::logTCPOpen(
+    StackdriverRootContext::TcpRecordInfo& record_info,
+    const ::Wasm::Common::FlatNode& peer_node) {
+  // Log TCP Open event log, if it has not been logged before.
+  if (!record_info.tcp_open_log_entry_logged) {
+    auto state = record_info.request_info->tcp_connection_state;
+    // It's possible that for a short lived TCP connection, we log TCP
+    // Connection Open log entry on connection close. To make a distinction
+    // between open and close log entries, we change the state to Open here.
+    record_info.request_info->tcp_connection_state =
```

Comment:
Does it make sense to always just set `tcp_connection_state` to Open in this function?

Reply:
That is so that we can capture `Connected` state too at timeout as this would tell if there was data sent on connection.. and not just `Open`...